src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -439,6 +439,139 @@ export abstract class AbstractSearchAndReplaceAction extends Action {
 
 		return previousElement;
 	}
+
+}
+class ReplaceActionRunner {
+	constructor(
+		private viewer: WorkbenchObjectTree<RenderableMatch>,
+		private viewlet: SearchView | undefined,
+		private getElementToFocusAfterRemoved: (viewer: WorkbenchObjectTree<RenderableMatch>, elementToBeRemoved: RenderableMatch) => RenderableMatch,
+		private getPreviousElementAfterRemoved: (viewer: WorkbenchObjectTree<RenderableMatch>, element: RenderableMatch) => RenderableMatch,
+		// Services
+		@IReplaceService private readonly replaceService: IReplaceService,
+		@IEditorService private readonly editorService: IEditorService,
+		@IConfigurationService private readonly configurationService: IConfigurationService,
+		@IUriIdentityService private readonly uriIdentityService: IUriIdentityService
+	) { }
+
+	async performReplace(element: FolderMatch | FileMatch | Match): Promise<any> {
+		// since multiple elements can be selected, we need to check the type of the FolderMatch/FileMatch/Match before we perform the replace.
+
+		const opInfo = getElementsToOperateOnInfo(this.viewer, element);
+
+		await Promise.all(opInfo.elementsToRemove.map(async (elem) => {
+			const parent = elem.parent();
+
+			if ((parent instanceof FolderMatch || parent instanceof FileMatch) && arrayContainsElementOrParent(parent, opInfo.elementsToRemove)) {
+				// skip any children who have parents in the array
+				return;
+			}
+
+			if (parent instanceof FolderMatch && elem instanceof FileMatch) {
+				await (<FolderMatch>(elem.parent())).replace(<FileMatch>elem);
+			} else if (parent instanceof FileMatch && elem instanceof Match) {
+				await (<FileMatch>(elem.parent())).replace(<Match>elem);
+			} else if (elem instanceof FolderMatch) {
+				await elem.replaceAll();
+			}
+		}));
+
+		const currentBottomFocusElement = opInfo.elementsToRemove[opInfo.elementsToRemove.length - 1];
+
+		if (currentBottomFocusElement instanceof Match) {
```

Comment:
I'm not sure why the type of the last element matters here?

Reply:
It's because, if you look at the original code, only `ReplaceAction`  runs this code where the replacepreview appears.

Reply:
Filed a bug for this: https://github.com/microsoft/vscode/issues/155879